enterprise/backend/test/metabase_enterprise/sso/integrations/jwt_test.clj

```diff
@@ -120,6 +120,33 @@
               redirect-url (get-in result [:headers "Location"])]
           (is (str/includes? redirect-url "&return_to=")))))))
 
+(deftest jwt-saml-both-enabled-test
+  (with-jwt-default-setup
+    (saml-test/with-saml-default-setup
+      (testing "with SAML and JWT configured, a GET request with JWT params should sign in correctly"
+        (let [response (saml-test/client-full-response :get 302 "/auth/sso"
+                                                       {:request-options {:redirect-strategy :none}}
+                                                       :return_to default-redirect-uri
+                                                       :jwt (jwt/sign {:email      "rasta@metabase.com"
+                                                                       :first_name "Rasta"
+                                                                       :last_name  "Toucan"
+                                                                       :extra      "keypairs"
+                                                                       :are        "also present"}
+                                                                      default-jwt-secret))]
+          (is (saml-test/successful-login? response))
+          (testing "redirect URI"
+            (is (= default-redirect-uri
+                   (get-in response [:headers "Location"]))))
+          (testing "login attributes"
+            (is (= {"extra" "keypairs", "are" "also present"}
+                   (t2/select-one-fn :login_attributes User :email "rasta@metabase.com"))))))
+
+      (testing "with SAML and JWT configured, a GET request without JWT params should redirect to SAML IdP"
+        (let [response (saml-test/client-full-response :get 302 "/auth/sso"
+                                                       {:request-options {:redirect-strategy :none}}
+                                                       :return_to default-redirect-uri)]
+          (is (not (saml-test/successful-login? response))))))))
```

Comment:
Why wouldn't this be a successful SAML login if both JWT & SAML are set up? Is it just because of the `request-options` being wrong?

Could the test assert that a SAML login can succeed when both are enabled in the `:jwt` key is omitted?

Reply:
It would redirect to the SAML login page in order to get the correct redirect params.